### PR TITLE
fix(azure_functions): run the mini agent regardless of azure hosting plan

### PR DESCRIPTION
**Pull request by gord02**
## What does this PR do?
Runs the Serverless Mini Agent for Azure Functions on all consumption plans not just if the hosting plan is consumption.

## Testing
The changes were tested by running the test suite for the internal folder using the following instructions for the repo: https://ddtrace.readthedocs.io/en/stable/contributing-testing.html#testing-guidelines

## Checklist

- [x] Change(s) are motivated and described in the PR description
- [x] Testing strategy is described if automated tests are not included in the PR
- [x] Risks are described (performance impact, potential for breakage, maintainability)
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed or label `changelog/no-changelog` is set
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/))
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))
- [x] If this PR changes the public interface, I've notified `@DataDog/apm-tees`.

## Reviewer Checklist

- [x] Title is accurate
- [x] All changes are related to the pull request's stated goal
- [x] Description motivates each change
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes
- [x] Testing strategy adequately addresses listed risks
- [x] Change is maintainable (easy to change, telemetry, documentation)
- [x] Release note makes sense to a user of the library
- [x] Author has acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

**Review by brettlangdon — COMMENTED** [review]
If this is a "fix" then it will need a release note.

File: ddtrace/_trace/tracer.py [hunk 1]
```diff
@@ -1150,11 +1150,7 @@ def _use_sync_mode() -> bool:
         - Google Cloud Functions and Azure Consumption Plan Functions have a mini-agent spun up by the tracer.
```
Comment (DylanLovesCoffee) [inline ddtrace/_trace/tracer.py:1150]: nit: might as well change this from Azure Consumption Plan Functions -> Azure Functions

**Review by DylanLovesCoffee — APPROVED** [review]
Looks good for Serverless

File: ddtrace/_trace/tracer.py [hunk 2]
```diff
@@ -549,9 +549,9 @@ def configure(
                 sync_mode=self._use_sync_mode(),
                 api_version=api_version,
                 # if apm opt out, neither agent or tracer should compute the stats
-                headers={"Datadog-Client-Computed-Stats": "yes"}
-                if (compute_stats_enabled or self._apm_opt_out)
-                else {},
+                headers=(
+                    {"Datadog-Client-Computed-Stats": "yes"} if (compute_stats_enabled or self._apm_opt_out) else {}
+                ),
```
Comment (mabdinur) [inline ddtrace/_trace/tracer.py:554]: Do we need to make this change?

File: ddtrace/internal/serverless/__init__.py [hunk 3]
```diff
@@ -31,17 +31,13 @@ def in_gcp_function():
...
     # type: () -> bool
-    """Returns whether the environment is an Azure Consumption Plan Function.
-    This is accomplished by checking the presence of two Azure Function env vars,
-    as well as a third SKU variable indicating consumption plans.
+    """Returns whether the environment is an Azure Function.
+    This is accomplished by checking the presence of two Azure Function env vars.
     """
     is_azure_function = (
         os.environ.get("FUNCTIONS_WORKER_RUNTIME", "") != "" and os.environ.get("FUNCTIONS_EXTENSION_VERSION", "") != ""
     )
```
Comment (mabdinur) [inline ddtrace/internal/serverless/__init__.py:42]: nit: We can simplify this function
```suggestion
    return os.environ.get("FUNCTIONS_WORKER_RUNTIME", "") != "" and os.environ.get("FUNCTIONS_EXTENSION_VERSION", "") != ""
```

File: ddtrace/internal/serverless/__init__.py [hunk 4]
```diff
@@ -31,17 +31,13 @@ def in_gcp_function():
...
 
-    website_sku = os.environ.get("WEBSITE_SKU", "")
-    is_consumption_plan = website_sku == "" or website_sku == "Dynamic"
-
-    return is_azure_function and is_consumption_plan
+    return is_azure_function
```
Comment (mabdinur) [inline ddtrace/internal/serverless/__init__.py:43]:
```suggestion
```

File: releasenotes/notes/allow-mini-agent-to-run-on-all-hosting-plans-5e8df7330c108d71.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+
+features:
+  - |
+    tracing: This removes the restrictions on the tracer to only run the mini-agent on the consumption plan. The mini-agent now runs regardless of the hosting plan
```
Comment (mabdinur) [inline releasenotes/notes/allow-mini-agent-to-run-on-all-hosting-plans-5e8df7330c108d71.yaml:4]: Add new line and nits
```suggestion
    tracing: Removes the restrictions on the tracer to only run the mini-agent on the consumption plan. The mini-agent now runs regardless of the hosting plan.

```

**Review by mabdinur — APPROVED** [review]
Left two nits but overall looks good to me

File: ddtrace/_trace/tracer.py (hunk 2 above)
Comment (gord02) [reply ddtrace/_trace/tracer.py:554]: I believe the formatting command was responsible for this. Should it be reverted?

Comment (mabdinur) [reply ddtrace/_trace/tracer.py:554]: Nope that's fine

File: releasenotes/notes/allow-mini-agent-to-run-on-all-hosting-plans-5e8df7330c108d71.yaml [hunk 6]
```diff
@@ -0,0 +1,5 @@
+
+features:
+  - |
+    tracing: Removes the restrictions on the tracer to only run the mini-agent on the consumption plan. The mini-agent now runs regardless of the hosting plan
+
```
Comment (mabdinur) [inline releasenotes/notes/allow-mini-agent-to-run-on-all-hosting-plans-5e8df7330c108d71.yaml:5]:
```suggestion

```

File: releasenotes/notes/allow-mini-agent-to-run-on-all-hosting-plans-5e8df7330c108d71.yaml [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+features:
+  - |
+    tracing: Removes the restrictions on the tracer to only run the mini-agent on the consumption plan. The mini-agent now runs regardless of the hosting plan
```
Comment (mabdinur) [inline releasenotes/notes/allow-mini-agent-to-run-on-all-hosting-plans-5e8df7330c108d71.yaml:4]:
```suggestion
    tracing: Removes the restrictions on the tracer to only run the mini-agent on the consumption plan. The mini-agent now runs regardless of the hosting plan

```

Comment (mabdinur) [inline releasenotes/notes/allow-mini-agent-to-run-on-all-hosting-plans-5e8df7330c108d71.yaml:4]:
```suggestion
   azure: Removes the restrictions on the tracer to only run the mini-agent on the consumption plan. The mini-agent now runs regardless of the hosting plan
```

File: ddtrace/internal/serverless/__init__.py [hunk 8]
```diff
@@ -31,17 +31,11 @@ def in_gcp_function():
...
 
 
-def in_azure_function_consumption_plan():
+def in_azure_function():
     # type: () -> bool
-    """Returns whether the environment is an Azure Consumption Plan Function.
-    This is accomplished by checking the presence of two Azure Function env vars,
-    as well as a third SKU variable indicating consumption plans.
+    """Returns whether the environment is an Azure Function.
+    This is accomplished by checking the presence of two Azure Function env vars.
```
Comment (erikayasuda) [inline ddtrace/internal/serverless/__init__.py:37]: nit:
```suggestion
    """Returns whether the environment is an Azure Function.
```

**Review by erikayasuda — APPROVED** [review]

**Pull request merged**